tests/sals/fs/test_fs.py

```diff
@@ -157,3 +163,178 @@ def test012_walk_dirs(self, recursive):
         else:
             self.assertEqual(1, len(dirs))
             self.assertNotIn(random_dir_dest_3, dirs)
+
+    def test013_path_parts(self):
+        random_dir_dest, random_dir_dest_2, _, _ = self.create_tree()
+        random_dir_dest_3 = j.sals.fs.join_paths(random_dir_dest_2, self.generate_random_text())
+        self.info("Create dir {}".format(random_dir_dest_3))
+        parts = list(j.sals.fs.path_parts(random_dir_dest_3))
+        split_path = random_dir_dest_3.split("/")
+        split_path[0] = "/"
+        self.assertEqual(parts, split_path)
+
+    def test014_basename(self):
+        random_dir_dest, random_dir_dest_2, _, _ = self.create_tree()
+        base =  self.generate_random_text()
+        random_dir_dest_3 = j.sals.fs.join_paths(random_dir_dest_2, base)
+        self.info("Create dir {}".format(random_dir_dest_3))
+
+        basename = j.sals.fs.basename(random_dir_dest_3)
+        self.info("Get Basename of  {}".format(random_dir_dest_3))
+        self.assertEqual(basename, base)
+
+    def test16_write_read_check_binary_file(self):
+        """
+        Test case for writting, reading and checking binary file.
+        **Test scenario**
+        #. Write binary file.
+        #. Check that file has been created and it is a binary file.
+        #. Read this file and check that its content.
+        """
+        self.info("Write binary file.")
+        content = self.random_string()
+        random_dir_dest, random_dir_dest_2, _, _ = self.create_tree()
+        random_dir_dest_3 = j.sals.fs.join_paths(random_dir_dest_2, self.generate_random_text())
+        with open(random_dir_dest_3, "wb") as f:
+            pickle.dump(content, f)
```

Comment:
Why do we use pickle here? we can write any binary content? and I think the implementation of binary file in this sal just means non-ascii.